### PR TITLE
Initial SSR e2e test

### LINKED ISSUE #12596: e2e suite for SSR
Suite of end to end tests for SSR validation

- [x] Validation of main pages
  - [x] Homepage
  - [x] PLP
  - [x] PDP
- [x] SEO elements validation
  - [x] canonical ulrs,
  - [x] title.
  - [x] description
  - [x] ld schema

- [ ] Add a parameter to make SSR timeout configurable

### PR DESCRIPTION
**Pull request by hackergil**
Fixes #12596

### CODE REVIEW

File: projects/storefrontapp/server.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { AppServerModule } from './src/main.server';
 // And we need to use esModuleInterop option in ssr dev mode, because i18next enforce usage of this option for cjs module.
 const express = require('express');
 
-const ngExpressEngine = NgExpressEngineDecorator.get(engine);
+const ngExpressEngine = NgExpressEngineDecorator.get(engine, null);
```
Comment (dunqan) [inline projects/storefrontapp/server.ts]: I'm in favor of keeping our SSR optimization engine enabled (as a part of the testing).
We can tweak the options thought.

Comment (hackergil) [reply projects/storefrontapp/server.ts]: I reverted this. Also, increasing the timeout by 2 seconds did the trick. Let me know if that's ok or we should do something else

File: package.json [hunk 2]
```diff
@@ -56,6 +56,7 @@
     "test:storefront:lib": "ng test storefrontlib --sourceMap --code-coverage",
     "dev:ssr": "env-cmd --no-override -e dev,b2c,$SPA_ENV cross-env NODE_TLS_REJECT_UNAUTHORIZED=0 ng run storefrontapp:serve-ssr",
     "serve:ssr": "node dist/storefrontapp-server/main.js",
+    "serve:ssr:ci": "NODE_TLS_REJECT_UNAUTHORIZED=0 SSR_TIMEOUT=0 node dist/storefrontapp-server/main.js",
```
Comment (dunqan) [inline package.json:59]: will it work cross-env? I mean Windows/Linux?

Comment (hackergil) [reply package.json:59]: It won't, but it's ok since it's only for CI and it does not run on windows. Would you agree?

**Review by dunqan — APPROVED** [review]

**Pull request merged**